apps/prairielearn/src/ee/lib/validateHTML.ts

```diff
@@ -281,6 +281,10 @@ function checkIntegerInput(ast: DocumentFragment | ChildNode): ValidationResult
           break;
         // string inputs are valid as strings, and these don't affect other tags, so no validation required
         case 'correct-answer':
+          if (val.match(mustacheTemplateExtractorRegex)){
+            errors.push(`pl-integer-input: correct-answer must be a static floating-point number, not dynamically generated in \`server.py\`. If the correct answer depends on internal parameters, simply set \`data[\'correct_answers\']\` accordingly and remove this attribute.`)
```

Comment:
I don't know if this is fully accurate, but I also can't imagine a single case where something is dynamically generated where we want to set it with correct-answer instead of setting it in server.py. Will replace with a python check in the very near future when we add python static checkers.